bpm/bonita-core/bonita-process-definition/bonita-process-definition-impl/src/test/java/org/bonitasoft/engine/core/process/definition/ProcessDefinitionServiceImplTest.java

```diff
@@ -360,6 +390,43 @@ public void getProcessDefinitionId_should_throw_SProcessDefinitionReadException_
         processDefinitionServiceImpl.getProcessDefinitionId("name", "version");
     }
 
+    @Test
+    public void getProcessDefinition_from_cache() throws Exception {
+        sProcessDefinitionDeployInfo.setLastUpdateDate(5478L);
+        final SProcessDefinitionImpl sProcessDefinition = new SProcessDefinitionImpl("a", "b");
+        doReturn(new Pair<Long, SProcessDefinition>(5478L, sProcessDefinition)).when(processDefinitionServiceImpl).getSProcessDefinitionFromCache(PROCESS_ID);
+
+        final SProcessDefinition processDefinition = processDefinitionServiceImpl.getProcessDefinition(PROCESS_ID);
+
+        assertThat(processDefinition).as("returned process definition from cache").isEqualTo(sProcessDefinition);
+        verify(cacheService, times(0)).store(anyString(), any(Serializable.class), any());
+    }
+
+    @Test
+    public void getProcessDefinition_from_database() throws Exception {
+        sProcessDefinitionDeployInfo.setLastUpdateDate(5478L);
+        final SProcessDefinitionImpl sProcessDefinition = new SProcessDefinitionImpl("a", "b");
+        doReturn(sProcessDefinition).when(processDefinitionServiceImpl).convertDesignProcessDefinition(designProcessDefinition);
+
+        final SProcessDefinition processDefinition = processDefinitionServiceImpl.getProcessDefinition(PROCESS_ID);
+
+        assertThat(processDefinition).as("returned process definition from database").isEqualTo(sProcessDefinition);
+        verify(cacheService, times(1)).store(anyString(), any(Serializable.class), any());
+    }
+
+    @Test
+    public void getProcessDefinition_from_database_when_cache_is_outdated() throws Exception {
+        sProcessDefinitionDeployInfo.setLastUpdateDate(5478L);
+        final SProcessDefinitionImpl processDefinitionB = new SProcessDefinitionImpl("a", "b");
+        final SProcessDefinitionImpl processDefinitionC = new SProcessDefinitionImpl("a", "c");
+        doReturn(new Pair<Long, SProcessDefinition>(5477L, processDefinitionB)).when(processDefinitionServiceImpl).getSProcessDefinitionFromCache(PROCESS_ID);
+        doReturn(processDefinitionC).when(processDefinitionServiceImpl).convertDesignProcessDefinition(designProcessDefinition);
+        final SProcessDefinition processDefinition = processDefinitionServiceImpl.getProcessDefinition(PROCESS_ID);
+
+        assertThat(processDefinition).as("returned process definition from cache").isEqualTo(processDefinitionC);
+        verify(cacheService, times(1)).store(anyString(), any(Serializable.class), any());
```

Comment:
Should you check that the call to the **store()** method is done with the right argument **processDefinitionC** ?